src/d_iwad.c

```diff
@@ -459,13 +459,15 @@ static boolean DirIsFile(char *path, char *filename)
 static char *CheckDirectoryHasIWAD(char *dir, char *iwadname)
 {
     char *filename; 
+    char *probe;
 
     // As a special case, the "directory" may refer directly to an
     // IWAD file if the path comes from DOOMWADDIR or DOOMWADPATH.
 
-    if (DirIsFile(dir, iwadname) && M_FileExists(dir))
+    probe = M_FileCaseExists(dir);
+    if (DirIsFile(dir, iwadname) && (probe != NULL))
     {
-        return M_StringDuplicate(dir);
+        return M_StringDuplicate(probe);
```

Comment:
`M_FileCaseExists()` allocates a new string, so no need for `M_StringDuplicate()` here too, correct? ie. this is a memory leak?